aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -291,7 +292,7 @@ public String getVersion() {
             try {
                 AppSyncGraphQLRequest<R> appSyncRequest = (AppSyncGraphQLRequest<R>) request;
                 AuthRule ownerRuleWithReadRestriction = null;
-                ArrayList<String> readAuthorizedGroups = new ArrayList<>();
+                Map<String, List<String>> readAuthorizedGroupsMap = new HashMap<>();
```

Comment:
The data type here is starting to look like a poor-man's data structure. Does it make sense to create a `class ReadAuthorizedGroups`, to encapsulate the data?

Reply:
> poor-man's data structure.
🤣  yep that's exactly right. I can do that, but I think that "map of lists" is a common enough data structure that it is passable as is. I think the gains from creating an entirely new class for this will be very little in terms of readability and intuitiveness. what do you think?

Reply:
Yea, it's not a big deal either way. Up to you. The benefit of doing the class is that you'll get to give a semantic label to the key and to the values. Looking at this `readAuthorizedGroupsMap` right now, I'm not certain what the key is, until I scroll down and look at the impl.

Reply:
Yeah I feel the benefit to seeing right away the structure of the data from the map outweighs the benefit of creating a custom class for naming purposes.